app/src/main/java/app/gamenative/service/gog/GOGDownloadManager.kt

```
@@ -776,182 +784,239 @@ class GOGDownloadManager @Inject constructor(
-                        retryResults.firstOrNull { it.isFailure }?.let { failedResult ->
-                            return@withContext Result.failure(
-                                failedResult.exceptionOrNull() ?: Exception("Failed to download chunk after link refresh"),
-                            )
+                        file.chunks.forEach { chunk ->
+                            if (!chunksAdded.contains(chunk.compressedMd5)) {
+                                chunksAdded.add(chunk.compressedMd5)
+                                networkChunkFlow.emit(chunk.compressedMd5)
+                                Timber.tag("GOG").v("Emitted chunk ${chunk.compressedMd5} to download flow")
+                            }
                         }
+                    } catch (e: IOException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)